benthomasson/eda/tests/integration/event_source_webhook/test_webhook_source.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+import os
+import requests
+import json
+from ..utils import CLIRunner, TESTS_PATH
+
+
+@pytest.fixture(scope="function")
+def factory_cli_runner():
+    """
+    Factory that returns an instance of the CLI with
+    teardown.
+    """
+    runners = []
+    def _cli_runner(env_vars: str = None, rules_file: str = "test_webhook_rules.yml"):
+        ruleset = os.path.join(TESTS_PATH, "event_source_webhook", rules_file)
+
+        runner = CLIRunner(
+            rules=ruleset, verbose=True, envvars=env_vars
+        ).run_in_background()
+
+        runners.append(runner)
+        return runner
+
+    yield _cli_runner
+    runners[0].terminate()
+
+
+@pytest.mark.parametrize(
+    "port",
+    [
+        pytest.param(5000, id="default_port"),
+        pytest.param(5001, id="custom_port")
+    ],
+)
+def test_webhook_source_sanity(factory_cli_runner, port: int):
+    """
+    Check the successful execution, response and shutdown
+    of the webhook source plugin.
+    """
+    msgs = [
+        json.dumps({"ping": "pong"}).encode("ascii"),
+        json.dumps({"shutdown": ""}).encode("ascii"),
+    ]
+
+    url = f"http://127.0.0.1:{port}/webhook"
+    headers = {"user-agent": "webhook_client/1.0.1"}
+
+    os.environ["WH_PORT"] = str(port)
+
+    cli_runner = factory_cli_runner(env_vars="WH_PORT")
+
+    for msg in msgs:
+        requests.post(url, data=msg, headers=headers)
+
+    result = cli_runner.stderr.read().decode()
+
+    assert "'msg': 'SUCCESS'" in result
+    assert headers["user-agent"] in result
+    assert f"'Host': '127.0.0.1:{port}'" in result
+
+    cli_runner.poll()
+    assert cli_runner.returncode == 0
+
+
+def test_webhook_source_error_handling(factory_cli_runner):
```

Comment:
If this test only check the case when the port it's already used, named it accordingly. For example: `test_webhook_source_with_busy_port`